src/panels/BasePanel.ts

```diff
@@ -124,10 +135,21 @@ function getMessageContext<TContent extends ContentId>(
         subscribeToMessages: (handler) => {
             webview.onDidReceiveMessage(
                 (message: object) => {
-                    if (!isValidMessage<ToVsCodeMessage<TContent>>(message)) {
+                    if (!isValidMessage<ToVsCodeMsgDef<TContent>>(message)) {
                         throw new Error(`Invalid message to VsCode: ${JSON.stringify(message)}`);
                     }
 
+                    const getTelemetryData = telemetryDefinition[message.command];
+                    if (getTelemetryData !== false) {
+                        const commandTelemetryData =
+                            getTelemetryData === true ? {} : getTelemetryData(message.parameters);
+                        const telemetryData = {
+                            ...commandTelemetryData,
+                            command: `${contentId}.${message.command}`,
```

Comment:
@peterbom It would be helpful if you can give a sample value or example of contentId and message.command just for others to know what will be logged/captured as part of telemetry.

Reply:
I've updated the PR description, but I'm guessing you mean in the code? Yes, that makes sense. Let me add...

Reply:
Updated

Reply:
@peterbom Thank you looks great !!